cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -1622,6 +1650,405 @@ pub(crate) fn emit(
             }
         }
 
+        Inst::CvtUint64ToFloatSeq {
+            to_f64,
+            src,
+            dst,
+            tmp_gpr1,
+            tmp_gpr2,
+        } => {
+            // Emit the following sequence:
+            //
+            //  cmp 0, %src
+            //  jl handle_negative
+            //
+            //  ;; handle positive, which can't overflow
+            //  cvtsi2sd/cvtsi2ss %src, %dst
```

Comment:
This is (indeed as Chris says) very clever.  Just a note that it won't work unchanged if we ever come to support X86_32, since `cvtsi2s{d,s}` will only take a 32 bit input in that case.

Reply:
Noted!